src/lib.rs

```diff
@@ -346,21 +355,29 @@ fn calculate_waste(
 }
 
 #[inline]
-fn calculate_fee(weight: u32, rate: f32) -> u64 {
-    (weight as f32 * rate).ceil() as u64
+fn calculate_fee(weight: u32, rate: f32) -> Result<u64, FeeError> {
+    if rate <= 0.0 {
+        Err(FeeError::NegativeFeeRate)
+    } else if rate > 1000.0 {
+        Err(FeeError::AbnormallyHighFee)
+    } else {
+        Ok((weight as f32 * rate).ceil() as u64)
+    }
 }
 
 /// Returns the effective value which is the actual value minus the estimated fee of the OutputGroup
 #[inline]
-fn effective_value(output: &OutputGroup, feerate: f32) -> u64 {
-    output
+fn effective_value(output: &OutputGroup, feerate: f32) -> Result<u64, FeeError> {
```

Comment:
Unrelated to this PR. But here it should be `long_term_fee_rate`? In that case, maybe makes sense to hardcode it at 10sats/vb like core? 

cc @delcin-raj